frigate/http.py

```diff
@@ -143,24 +143,64 @@ def delete_retain(id):
         event = Event.get(Event.id == id)
     except DoesNotExist:
         return make_response(
-            jsonify({"success": False, "message": "Event" + id + " not found"}), 404
+            jsonify({"success": False, "message": "Event " + id + " not found"}), 404
         )
 
     event.retain_indefinitely = False
     event.save()
 
     return make_response(
-        jsonify({"success": True, "message": "Event" + id + " un-retained"}), 200
+        jsonify({"success": True, "message": "Event " + id + " un-retained"}), 200
     )
 
+@bp.route("/events/<id>/sub_label", methods=("POST",))
+def set_sub_label(id):
+    try:
+        event = Event.get(Event.id == id)
+    except DoesNotExist:
+        return make_response(
+            jsonify({"success": False, "message": "Event " + id + " not found"}), 404
+        )
+
+    if request.json:
+        new_sub_label = request.json.get("subLabel", "")
+    else:
+        new_sub_label = ""
+
+
+    if new_sub_label:
+        event.sub_label = new_sub_label
+        event.save()
+        return make_response(
+            jsonify({"success": True, "message": "Event " + id + " sub label set to " + new_sub_label}), 200
+        )
+
+    return make_response(
+        jsonify({"success": False, "message": "Sublabel " + new_sub_label + " not provided."}), 400
+    )
+
+@bp.route("/events/<id>/sub_label", methods=("DELETE",))
+def delete_sub_label(id):
```

Comment:
Technically you don't need this route because posting to the other endpoint without providing a value will set it back to None too.

Reply:
Makes sense, I had the other route guarded so providing None would result in nothing happening, but I will adjust that behavior so only one route is necessary